flink-table/flink-table-planner/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -0,0 +1,5161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.tools;
```

Comment:
We offer a `FlinkRelBuilder` already. Can we override the affected methods instead of copying the entire `RelBuilder` class? It seems only the

```
public RelBuilder join(JoinRelType joinType, RexNode condition,
      Set<CorrelationId> variablesSet) {
```

Is affected?

Reply:
yes, only this method is affected.
The problem with `FlinkRelBuilder` is that `org.apache.calcite.tools.RelBuilder#join(org.apache.calcite.rel.core.JoinRelType, org.apache.calcite.rex.RexNode, java.util.Set<org.apache.calcite.rel.core.CorrelationId>)` operates with private fields `stack`, `simplifier`, `config`, `struct` and private static classes `org.apache.calcite.tools.RelBuilder.Frame`, `org.apache.calcite.tools.RelBuilder.Shifter` which are not reachable from `FlinkRelBuilder`... 
At the same time probably it could be a point to improve it on Calcite side to make it extendable in future

Reply:
I added a follow up task for that https://issues.apache.org/jira/browse/CALCITE-5381
Once it is fixed this class could be removed
By the way: I've checked it till 1.32.0 and this class does not require modification from Flink to make it working